plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java

```diff
@@ -714,6 +715,14 @@ public void testArithmeticPredicatePushdown()
         assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % nationkey = 2"))
                 .isFullyPushedDown()
                 .matches("VALUES (BIGINT '3', CAST('CANADA' AS varchar(25)), BIGINT '1')");
+
+        // some databases calculate remainder instead of modulus when one of the values is negative
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % -nationkey = 2"))
+                .isFullyPushedDown()
+                .matches("VALUES (BIGINT '3', CAST('CANADA' AS varchar(25)), BIGINT '1')");
+
+        assertThatThrownBy(() -> query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % 0 = 2"))
+                .hasMessageContaining("ERROR: division by zero");
```

Comment:
That could be BCT.

Also, what if we have `% (regionkey-1)` (a variable on the RHS) ?

Reply:
Good point, I'll add a test.

Reply:
Added, but kept in TestPostgreSqlConnectorTest for now since most connectors don't support connector expression pushdown yet.

Reply:
We can have correctness-only coverage in BCT.
or guard this with `SUPPORTS_PREDICATE_EXPRESSION_PUSHDOWN`